xbmc/platform/linux/storage/UDisks2Provider.h

```diff
@@ -0,0 +1,150 @@
+#pragma once
+/*
+ *      Copyright (C) 2005-2013 Team XBMC
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <string>
+#include <vector>
+
+#include "storage/IStorageProvider.h"
+#include "DBusUtil.h"
+
+class CUDisks2Drive
+{
+public:
+  std::string m_object;
```

Comment:
I'd made them private and rather the other Udiskclasses that directly use it friend classes to not provide direct access to everyone.

Reply:
I found it more explicit to turn helper classes into nested classes. This also allows uncluttering the class names and to turn the non-member helper funcion into a static member.

Please check 252724db449027b57817cb2289519aa3f30e4dbf